frame/clearing-house/src/lib.rs

```diff
@@ -652,18 +638,13 @@ pub mod pallet {
 			amount: Self::Balance,
 		) -> Result<(), DispatchError> {
 			ensure!(
-				CollateralTypes::<T>::contains_key(asset_id),
+				CollateralType::<T>::get().ok_or(Error::<T>::NoCollateralTypeSet)? == asset_id,
 				Error::<T>::UnsupportedCollateralType
 			);
 
 			// Assuming stablecoin collateral and all markets quoted in dollars
-			T::Assets::transfer(
-				asset_id,
-				account_id,
-				&T::PalletId::get().into_account(),
-				amount,
-				true,
-			)?;
+			let pallet_acc = T::PalletId::get().into_sub_account("Collateral");
```

Comment:
I would move this into its own function called get_collateral_account or something similar